iml-fs/src/remove_dir_all.rs

```diff
@@ -0,0 +1,67 @@
+use futures::{future::Future, Async, Poll};
+use std::{fs, io, path::Path, pin::Pin};
+use tokio_threadpool::blocking;
+
+/// tokio-fs.remove_dir_all
+
+/// Removes a directory at this path, after removing all its contents. Use carefully!
+///
+/// This is an async version of [`std::fs::remove_dir_all`][std]
+///
+/// [std]: https://doc.rust-lang.org/std/fs/fn.remove_dir_all.html
+pub fn remove_dir_all<P: AsRef<Path>>(path: P) -> RemoveDirAllFuture<P> {
+    RemoveDirAllFuture::new(path)
+}
+
+/// Future returned by `remove_dir_all`.
+#[derive(Debug)]
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+pub struct RemoveDirAllFuture<P>
+where
+    P: AsRef<Path>,
+{
+    path: P,
+}
+
+impl<P> RemoveDirAllFuture<P>
+where
+    P: AsRef<Path>,
+{
+    fn new(path: P) -> RemoveDirAllFuture<P> {
+        RemoveDirAllFuture { path }
+    }
+}
+
+///our future FeaturesFuture
+impl<P> Future for RemoveDirAllFuture<P>
+where
+    P: AsRef<Path>,
+{
+    type Item = ();
+    type Error = io::Error;
+
+    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+        blocking_io(|| fs::remove_dir_all(&self.path))
+    }
+}
+
+/// tokio-fs.lib
+fn blocking_io<F, T>(f: F) -> Poll<T, io::Error>
```

Comment:
This is unnecessary as blocking_io is already part of tokio 0.1.

Take a look at https://github.com/tokio-rs/tokio/blob/59fb5b9a7d89ee84389b8f8a039e874ef8971dbd/tokio-fs/src/create_dir.rs for example of how this can be done.

Reply:
But this function is private

Reply:
Ah, that's too bad. 

So, Instead of carrying tokio internals locally, I'd argue we should create a local version that has similar semantics but is much simpler.

```
pub fn remove_dir_all<P: AsRef<Path>>(p: P) -> impl Future<Item = (), Error = io::Error> {
    poll_fn(move || {
        blocking(|| fs::remove_dir_all(&p)).map_err(|_| panic!("the threadpool shut down"))
    })
    .and_then(std::convert::identity)
}
```

We should still submit an upstream patch to tokio 0.1.x, and once that lands we should remove this fn.